cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/CruiseControlReconciler.java

```diff
@@ -249,6 +251,8 @@ protected Future<Void> apiSecret() {
                             newSecret.setData(oldSecret.getData());
                         }
 
+                        this.apiCredentialsHash = String.valueOf(newSecret.hashCode());
```

Comment:
How does it work when you hash the whole Secret object? Doesn't it just hash some default `toString` method result? I wonder if it would be better to hash some proper Strings with the values.

Reply:
This is using the hashCode method generated by Lombok, which uses all non-static, non-transient fields. I think this is fine, but I don't have strong opinion. We can join all values and hash that if you prefer.

Reply:
I think it would be good to have at least some better unit tests that make sure that the value does not change between different reconciliations where the new Secrets are generated. For example, check for the exact hash in the test you have instead of `notNullValue`. Because this depends on the implementation that we do not have under control. I also wonder how much this exposes us to changes to the underlying Fabric8 objects that we might not care about (while we might not care about them in most cases as the pods anyway roll at operator upgrade, it might _surprise_ us one day for some reason).

Reply:
Ok, makes sense.